documentation/smg/17_pixel_data.md

```diff
@@ -0,0 +1,397 @@
+            while obj.has_more()
+                % Move to the next page of data.
+                % Since the current page has been edited, this 'advance' dumps
+                % the page to a temporary file and marks it as "dirty".
+                % It then clears the cache, and increments the internally held
+                % `page_number`.
+                obj.advance();
+
+                % This call will now load the data for `page_number` and
+                % perform the subtraction.
+                obj.signal = obj.signal - scalar;
+            end
+        end
+
+        ...
+
+    end
+end
+```
+
+### Managing Binning
+
+Some algorithms require that pixels in different bins are operated on in
+different ways.
+In most cases, bins will cross pixel page boundaries,
+so it takes some work to find which bins require which action.
+
+As an example, consider performing addition between an `sqw` and a `dnd` object.
+Each pixel in the `sqw` object, that belongs to bin `n`,
+should have its signal increased by the value of the `dnd` object's signal in
+bin `n`.
+
+To make these cases easier, the function `split_vector_fixed_sum` was written.
+This function will split the `npix` array
+(the array that defines the pixel binning)
+into chunks such that each chunk defines the binning for a page of pixel data.
+
+```matlab
+classdef PixelData < handle
+
+    methods
+
+        ...
+
+        function binary_op_sigvar(obj, sigvar_obj, npix)
+            %BINARY_OP_SIGVAR Add a sigvar-like object to this pixel data
+            %  This is a simplified version of `binary_op_sigvar_` that exists
+            %  in Horace (we don't worry about error in this version).
+            %
+            %  sigvar-like refers to an object that has a signal `.s` and
+            %  error `.e`.
+            %
+
+            % Always move to the first page of data
+            obj.move_to_first_page();
+
+            % The given npix array defines the number of pixels in each image
+            % bin.
+            % Split this array up on page size, such that we get the number of
+            % pixels in each bin for each page of pixel data.
+            % `img_idxs` is a [2xN] array where N is the number of npix chunks,
+            % and each column is the start and end index of the range of bins
+            % the npix chunk defines the binning of.
+            [npix_chunks, img_idxs] = split_vector_fixed_sum(npix, obj.base_page_size);
+
+            for page_number = 1:numel(npix_chunks)
+                % Get the signal at the bins for which the currently loaded
+                % page of pixels contribute
+                img_chunk = sigvar_obj.s(img_idxs(1, page_number):img_idxs(2, page_number);
+
+                % "Smear" the sigvar image's signal, such that it has the same
+                % number of elements as we have pixels in the current page.
+                signal_to_add = repelem(img_chunk(:), npix_chunks{page_number});
```

Comment:
Sorry, but I do not understand this. Vaguely it appears you are getting img_chunk out of sigvar_obj.s as the values for the current page. I do not understand why this is not added directly to obj.signal. I do not understand why img_chunk needs to be put through repelem. I do not understand what smearing means. It would be good if you could provide more explicit explanation here. Thanks

Reply:
I've added more detail to the problem we're trying to solve in the function header. Hopefully that makes it more understandable, but let me know if it needs further clarification.